src/Elasticsearch/ClientBuilder.php

```diff
@@ -197,7 +197,8 @@ public static function fromConfig(array $config, bool $quiet = false): Client
     {
         $builder = new static;
         foreach ($config as $key => $value) {
-            $method = "set$key";
+            $allowedMethods = ['includePortInHostHeader']; // other available methods
```

Comment:
Might it be better to move this hard-coded list of allowed methods up to class scope? It would be potentially easier to find there, and wouldn't get created for each loop iteration.

Reply:
Good suggestion, I'll move it using a class constant.